MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs

```diff
@@ -827,13 +808,56 @@ internal void CreateSizeDependentResources()
             };
         }
 
+        
+
         internal void OnPresentationChanged()
         {
             CreateSizeDependentResources();
             ApplyRenderTargets(null);
         }
 
 #endif // WINDOWS
+        /// <returns>Higher than zero if multiSampleCount is supported. 
+        /// Zero if multiSampleCount is not supported.</returns>
+        private int GetMultiSamplingQuality(Format format, int multiSampleCount)
+        {
+            // The valid range is between zero and one less than the level returned by CheckMultisampleQualityLevels
+            // https://msdn.microsoft.com/en-us/library/windows/desktop/bb173072(v=vs.85).aspx
+            var quality = _d3dDevice.CheckMultisampleQualityLevels(format, multiSampleCount) - 1;
+            // NOTE: should we always return highest quality?
+            return Math.Max(quality, 0); // clamp minimum to 0 
+        }
+        private SampleDescription GetSupportedSampleDescription(Format format)
+        {
+            var multisampleDesc = new SharpDX.DXGI.SampleDescription(1, 0);
+
+
+            if (PresentationParameters.MultiSampleCount > 1)
+            {
+                // Round down MultiSampleCount to the nearest power of two
+                // hack from http://stackoverflow.com/a/2681094
+                // Note: this will return an incorrect, but large value
+                // for very large numbers. That doesn't matter because
+                // the number will get clamped below anyway in this case.
+                // Also MultiSampleCount should never be larger than 32 
+                // (which is MultiSampleCountLimit)
+                var msc = PresentationParameters.MultiSampleCount;
+                msc = msc | (msc >> 1);
+                msc = msc | (msc >> 2);
+                msc = msc | (msc >> 4);
+                msc -= (msc >> 1);
+                // and clamp it to what the device can handle
+                if (msc > GraphicsCapabilities.MaxMultiSampleCount)
+                    msc = GraphicsCapabilities.MaxMultiSampleCount;
+
+                var quality = GetMultiSamplingQuality(format, msc);
+
+                multisampleDesc.Count = msc;
+                multisampleDesc.Quality = quality;
+            }
+
+            return multisampleDesc;
+        }
```

Comment:
Since the clamping is platform agnostic it could be put in the shared GraphicsDevice.cs file so other platforms can reuse it. (wihtou the SampleDescription obviously)